src/Corax/Indexing/IndexWriter.cs

```diff
@@ -687,28 +685,50 @@ private void ResetWriter()
             _tempListBuffer = new NativeIntegersList(_entriesAllocator);
             _termsPerEntryId = new NativeList<NativeList<RecordedTerm>>();
             _termsPerEntryIds = new NativeList<long>();
-
+            _numberOfModifications = 0;
+            _numberOfTermModifications = 0;
+            _initialNumberOfEntries = _indexMetadata?.ReadInt64(Constants.IndexWriter.NumberOfEntriesSlice) ?? 0;
             _pforDecoder = new FastPForDecoder(_entriesAllocator);
         }
 
-        public void TryDeleteEntryByField(string field, string term)
+        public void TryDeleteEntryByField(string field, string term, long? numberOfModifications = null)
         {
-            using var __ = Slice.From(_entriesAllocator, term, ByteStringType.Immutable, out var termSlice);
-            using var ___ = Slice.From(_entriesAllocator, field, ByteStringType.Immutable, out var fieldSlice);
-            if (TryGetEntryTermId(fieldSlice, termSlice.AsSpan(), out long idInTree) == false) 
+            if (_fieldsMapping.TryGetByFieldName(field, out var binding) && binding.FieldId == 0)
+            {
+                TryDeleteEntry(term);
                 return;
+            }
+
+            long idInTree;
+            using (Slice.From(_entriesAllocator, term, ByteStringType.Immutable, out var termSlice))
+            using (Slice.From(_entriesAllocator, field, ByteStringType.Immutable, out var fieldSlice))
+            {
+
+                if (TryGetEntryTermId(fieldSlice, termSlice.AsSpan(), out idInTree) == false)
+                    return;
+            }
 
-            RecordDeletion(idInTree, out _);
+            numberOfModifications ??= _numberOfModifications;
+            RecordDeletion(idInTree, out _, out var notFoundDuplicatesInItemsToRemove);
+            if (notFoundDuplicatesInItemsToRemove == false)
+            {
+                FlushBatch();
+                TryDeleteEntryByField(field, term, numberOfModifications);
+            }
+            
         }
-  
         
         /// <summary>
         /// Record term for deletion from Index.
         /// </summary>
         /// <param name="idInTree">With frequencies and container type.</param>
+        /// <param name="notFoundDuplicatesInItemsToRemove">Intersection between PostingList and _deletedEntries</param>
         [SkipLocalsInit]
-        private bool RecordDeletion(long idInTree, out long singleEntryId)
+        private bool RecordDeletion(long idInTree, out long singleEntryId, out bool notFoundDuplicatesInItemsToRemove)
```

Comment:
I would be easier to follow the code if the polarity of `notFoundDuplicatesInItemsToRemove` got changed so we'd have `foundDuplicatesInItemsToRemove`. 

Please consider renaming it to `foundInEntriesToDelete` or `alreadyScheduledForDeletion` maybe?

Reply:
Changed into `setsAreDisjoints`. It's used to indicate if we have any document already scheduled for removal.

Reply:
Shouldn't be "disjoint" instead of "disjoints"?